service-data-access/build.gradle

```
@@ -36,31 +44,63 @@ dependencies {
-    testImplementation 'org.testcontainers:rabbitmq:1.17.3'
     testImplementation 'org.apache.camel.springboot:camel-spring-boot-starter:3.18.0'
     testImplementation 'org.apache.camel.springboot:camel-rabbitmq-starter:3.18.0'
     testImplementation 'org.apache.camel.springboot:camel-jackson-starter:3.18.0'
     testImplementation 'org.skyscreamer:jsonassert:1.2.3'
+    testImplementation 'org.mockito:mockito-junit-jupiter:4.7.0'
+
+    testImplementation 'org.junit.jupiter:junit-jupiter-engine:5.9.0'
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.9.0'
+
+
+    integrationTestImplementation 'org.testcontainers:testcontainers:1.17.3'
+    integrationTestImplementation 'org.testcontainers:junit-jupiter:1.17.3'
+    integrationTestImplementation 'org.testcontainers:rabbitmq:1.17.3'
+    integrationTestImplementation 'org.springframework.boot:spring-boot-starter-test:2.6.7'
+    integrationTestImplementation 'org.testcontainers:testcontainers:1.17.3'
+    integrationTestImplementation 'org.testcontainers:junit-jupiter:1.17.3'
+    integrationTestImplementation 'org.testcontainers:rabbitmq:1.17.3'
+    integrationTestImplementation 'org.apache.camel.springboot:camel-spring-boot-starter:3.18.0'
+    integrationTestImplementation 'org.apache.camel.springboot:camel-rabbitmq-starter:3.18.0'
+    integrationTestImplementation 'org.apache.camel.springboot:camel-jackson-starter:3.18.0'
+    integrationTestImplementation 'org.skyscreamer:jsonassert:1.2.3'
 }
 
-docker {
-    files "build/libs/service-data-access-0.0.1-SNAPSHOT.jar", "bin/entrypoint.sh"
-    buildArgs([JAR_FILE: "service-data-access-0.0.1-SNAPSHOT.jar", ENTRYPOINT_FILE: "entrypoint.sh"])
+configurations {
+    integrationTestImplementation.extendsFrom implementation
+    integrationTestRuntimeOnly.extendsFrom runtimeOnly
 }
 
-//dockerRun {
-//    env 'SECRETHUB_HELLO': getEnvOrDefault('SECRETHUB_HELLO', 'override-me'),
-//            'JAVA_PROFILE': '-Dspring.profiles.include=docker'
-//}
+jacoco {
+    toolVersion = "0.8.8"
+    reportsDirectory = file("$buildDir/jacoco")
+}
 
+test {
+    useJUnitPlatform()
+    exclude '**/integrationTest'
+    finalizedBy jacocoTestReport {
+        executionData(file("${project.buildDir}/jacoco/test.exec"))
+    }
+}
 
-// application {
-//     // Define the main class for the application.
-//    mainClass = 'gov.va.vro.abd_data_access.AbdApplication'
-// }
+task integrationTest(type: Test) {
+    description = 'Runs the integration tests.'
+    useJUnitPlatform()
+    testClassesDirs = sourceSets.integrationTest.output.classesDirs
+    classpath = sourceSets.integrationTest.runtimeClasspath
+    finalizedBy jacocoTestReport {
+        executionData(file("${project.buildDir}/jacoco/integrationTest.exec"))
+    }
+}
+
+docker {
+    files "build/libs/service-data-access-${project.version}.jar", "bin/entrypoint.sh"
+    buildArgs([JAR_FILE: "service-data-access-${project.version}.jar", ENTRYPOINT_FILE: "entrypoint.sh"])
```

```suggestion
    files "build/libs/${bootJar.archiveFileName.get()}", "bin/entrypoint.sh"
    buildArgs([JAR_FILE: bootJar.archiveFileName.get(), ENTRYPOINT_FILE: "entrypoint.sh"])
```

See https://github.com/department-of-veterans-affairs/abd-vro/blob/c6aefb88480907c74ced6a7013381cbdb98a67e7/buildSrc/README.md#starterjavacontainer-conventionsgradle

---

Good